src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -0,0 +1,210 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlServer.Management.SqlParser.Intellisense;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Workspace.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.LanguageServices
+{
+    /// <summary>
+    /// Peek Definition/ Go to definition implementation
+    /// Script sql objects and write create scripts to file
+    /// </summary>
+    internal class PeekDefinition
+    {
+        private ConnectionInfo connectionInfo;
+        private string tempPath;
+        
+        // Dictionary that holds the script getter for each type
+        private Dictionary<DeclarationType, Func<string, string, StringCollection>> sqlScriptGetters =
+            new Dictionary<DeclarationType, Func<string, string, StringCollection>>();
+
+        //Dictionary that holds the object name (as appears on the TSQL create statement)
+        private Dictionary<DeclarationType, string> sqlObjectTypes = new Dictionary<DeclarationType, string>();
+
+        private Database database 
+        {
+            get
+            {
+                Server server = new Server(this.connectionInfo.SqlConnection.DataSource);
+                return server.Databases[this.connectionInfo.SqlConnection.Database];
+            }
+        }
+        
+        internal PeekDefinition(ConnectionInfo connInfo)
+        {
+            connectionInfo = connInfo;
+            DirectoryInfo tempScriptDirectory = Directory.CreateDirectory( Path.GetTempPath()+ "mssql_definition");
+            tempPath = tempScriptDirectory.FullName;
+            Initialize();
+        }
+
+        /// <summary>
+        /// Add getters for each sql object supported by peek definition
+        /// </summary>
+        private void Initialize()
+        {
+            //Add script getters for each sql object
+
+            //Add tables to supported types
+            sqlScriptGetters.Add(DeclarationType.Table, GetTableScripts);
+            sqlObjectTypes.Add(DeclarationType.Table, "Table");
```

Comment:
When doing the codegen work you'll likely want to make a method that adds to both of these dictionaries in 1 operation. This will make it easier since there's less code and we guarantee that the value is in both dictionaries (which is assumed later on, but right now requires everyone to do the right thing when declaring a new handler)